### PR TITLE
feat: prepare floor planner

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR prepares for [FloorPlanner](https://github.com/kroma-network/halo2/tree/dev/halo2_proofs/src/circuit/floor_planner).
This implements [Circuit](https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/plonk/circuit.rs#L665-L687) and [Layouter](https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/circuit.rs#L410-L494)

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/circuit/assignment.h [hunk 1]
```diff
@@ -105,12 +105,14 @@ class Assignment {
 
   // Creates a new (sub)namespace and enters into it.
   //
+  // TODO(chokobole): Update comment since NamespacedLayouter comes in.
```
Comment (TomTaehoonKim) [inline tachyon/zk/plonk/circuit/assignment.h:108]: I don't get the comment. Did you mean
```suggestion
  // TODO(chokobole): Update comment when NamespacedLayouter comes in.
```
? To remind in the future?
If my guess is correct, could you please fix all the comments below as well?

Comment (chokobole) [reply tachyon/zk/plonk/circuit/assignment.h:108]: Oh yeap you're correct! I'll fix them al.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Comment by chokobole** [comment]
I'll merge this since the CI failure is not because of code change.

**Pull request merged**